src/charm.py

```diff
@@ -66,25 +66,26 @@ def version(self) -> Optional[str]:
 
 
 class AlertmanagerCharm(CharmBase):
-    _container_name: str = "alertmanager"  # automatically determined from charm name
-    _layer_name = "alertmanager"  # layer label argument for container.add_layer
-    _service_name: str = "alertmanager"  # chosen arbitrarily to match charm name
-    _peer_relation_name: str = "replicas"  # must match metadata.yaml peer role name
-    _api_port: int = 9093  # port to listen on for the web interface and API
-    _ha_port: int = 9094  # port for HA-communication between multiple instances of alertmanager
+    # Container name is automatically determined from charm name
+    # Layer name is used for the layer label argument in container.add_layer
+    # Service name is chosen arbitrarily to match charm name
+    _container_name = _layer_name = _service_name = "alertmanager"
+    _peer_relation_name = "replicas"  # must match metadata.yaml peer role name
+    _api_port = 9093  # port to listen on for the web interface and API
```

Comment:
Why are these not in `config.yaml`?

Reply:
The port number used to be, but since it is internal to the container it shouldn't matter for the user.

Reply:
I don’t agree with this assumption. Even if administrators may not change this often, once the nginx-ingress operator supports [arbitrary TCP/UCP remapping](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/exposing-tcp-udp-services.md), having this information at least visible (even if it’s not configurable) will be useful for administrators via something like `service-port` in the [existing charm](https://charmhub.io/nginx-ingress-integrator) if they want to expose it. 

This is a small change now (add it back to `config.yaml` and use `self.model.config[“port”]` instead) to future proof and save yourself headache later

Reply:
Captured in https://github.com/canonical/alertmanager-operator/issues/19